src/diffusers/loaders.py

```diff
@@ -2276,6 +2436,93 @@ def enable_lora_for_text_encoder(self, text_encoder: Optional[PreTrainedModel] =
             raise ValueError("Text Encoder not found.")
         set_adapter_layers(self.text_encoder, enabled=True)
 
+    def set_adapters(
+        self,
+        adapter_names: Union[List[str], str],
+        unet_weights: List[float] = None,
+        te_weights: List[float] = None,
+        te2_weights: List[float] = None,
+    ):
+        # Handle the UNET
+        self.unet.set_adapters(adapter_names, unet_weights)
+
+        # Handle the Text Encoder
+        if hasattr(self, "text_encoder"):
+            self.set_adapters_for_text_encoder(adapter_names, self.text_encoder, te_weights)
+        if hasattr(self, "text_encoder_2"):
+            self.set_adapters_for_text_encoder(adapter_names, self.text_encoder_2, te2_weights)
```

Comment:
Could we just have `te_weights()`? I think we can collate weights for multiple text encoders and operate on them, no? 

Here's a reference: https://github.com/huggingface/diffusers/blob/main/src/diffusers/loaders.py#L2056C1-L2067C62. 

@patrickvonplaten WDYT?

Reply:
I think currently this is fine as that way we don't have to re-define `set_adapters` in `StableDiffusionXLLoraLoaderMixin` , let me know if that does makes sense

Reply:
If I understand it correctly, it's already just `text_encoder_weights`

https://github.com/huggingface/diffusers/blob/4ac205e32fc44b6d92488304d74781a83945b9e6/src/diffusers/loaders.py#L2226

What am I missing out on?

Reply:
Yes but `set_adapters_for_text_encoder` takes only a single `text_encoder` as an input no? How would you call that method once for all the text encoders?

Reply:
Yes, that makes sense. Thanks for explaining.